lib/k8s/client.ex

```diff
@@ -22,7 +22,7 @@ defmodule K8s.Client do
   @type option :: {:name, String.t()} | {:namespace, binary() | :all}
   @type options :: [option]
 
-  alias K8s.Operation
+  alias K8s.Client.{Declarative, Imperative}
```

Comment:
I don't know that I'm totally sold on this dichotomy needing to exist or being desirable, or on including it in this PR. I think the new `apply` function could've lived wherever and that a refactor should've been a separate issue/PR, because now Cory has to decide whether to reject some of your changes while valuing others, and either accepting the subjective along with the good or forcing you to re-work. Part of being neighborly in OSS is working to make your PRs transparent and reviewer-friendly. Just my $0.02.

Reply:
Fair enough. What do you think, @coryodaniel? How do you want to handle this?